episodes/2-keras.Rmd

```diff
@@ -735,17 +740,16 @@ visualize a confusion matrix.
 
 :::: solution
 ## Solution
-The confusion matrix shows that the predictions for Adelie and Gentoo
-are decent, but could be improved. However, Chinstrap is not predicted
-ever.
-The training loss was very low, so from that perspective this may be
-surprising.
-But this illustrates very well why a test set is important when training
-neural networks.
-We can try many things to improve the performance from here.
-One of the first things we can try is to balance the dataset better.
-Other options include: changing the network architecture or changing the
-training parameters
+The confusion matrix shows that the predictions for Adelie and Gentoo are decent, but could be improved. However, Chinstrap is not predicted ever.
+Why the chinstraps are not present in the the predictions in the confusion matrix?
+If we go back to the [**Pair Plot**](https://carpentries-incubator.github.io/deep-learning-intro/2-keras.html#pair-plot) in the Visualization section, we can figure it out that the biggest challenge is masking the Chinstrap penguins from the marginal distributions of the four features (bill length, bill depth, flipper length, and body mass). That is, There is no single variable that separates Chinstrap from all other species. The combination of bill_length_mm vs. bill_depth_mm (the 2nd subfigure in the first row in the pairplot) gives a good separation of Chinstrap from Adelie and Gentoo.
+However, as there are limited number combinations that can be used for a clear classification of Penguins species, the constructed neural network (one dense layer and 10 neurons in this layer) might be difficult to generate a clear classification of Penguins species.
+A practical workaround is to increase the number of dense layers and also the number of neutrons in each dense layers. In addition, adjusting the learning rate is also possible to achive a high score for the prediction. You will get more info in the [**Advanced layer types**](https://carpentries-incubator.github.io/deep-learning-intro/4-advanced-layer-types.html) episode.
+
+The training loss was very low, so from that perspective this may be surprising. But this illustrates very well why a test set is important when training neural networks.
```

Comment:
```suggestion
The training loss was very low, so from that perspective this may be surprising. But this illustrates very well why a test set is important to give a realistic evaluation when training neural networks.
```